src/components/colors-legend.module.scss

```diff
@@ -14,13 +14,50 @@
   }
 
   .legendText {
-    max-width: 250px;
     overflow: hidden;
     text-overflow: ellipsis;
+    white-space: nowrap;
+    transition: width 0.27s ease-out;
+    width: 170px;
+
+    @media screen and (min-width: 360px) {
+      width: 200px;
+    }
+
+    @media screen and (min-width: 400px) {
+      width: 250px;
+    }
+  }
+
+  .collapsedText {
+    width: 0;
+  }
+
+  .legendIcon {
+    min-width: 35px;
   }
 }
 
 .legendContainer {
   max-height: 300px;
   overflow-y: auto;
+  padding-left: 40px;
+}
+
+.collapseWrapper {
+  position: absolute;
+  left: 0;
+  top: 0;
+  height: 100%;
+  width: 40px;
```

Comment:
CSS is my weakness and a layout like this confuses me.
Is it possible to not use `padding-left` and `position: absolute;` here?

Maybe:

```css
.legendContainer {
  display: flex;
  align-items: center;
```

Reply:
trying this way, I get stuck with 100% height for the child when the parent doesn't have a fixed height but only max-height. Do you have some suggestions on how to fix it?

Reply:
Strange !!!!
```css
display: flex;
align-items: center;
```
should do the work.
Maybe there are some issues with the collapseWrapper. Did you also apply `align-items: center;` for it?